DuckDB.NET.Data/DuckDBSchema.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Linq;
+using System.Text;
+
+namespace DuckDB.NET.Data;
+
+internal static class DuckDBSchema
+{
+    public static DataTable GetSchema(DuckDBConnection connection, string collectionName, string?[]? restrictionValues) =>
+        collectionName.ToUpperInvariant() switch
+        {
+            "METADATACOLLECTIONS" => GetMetaDataCollections(),
+            "RESTRICTIONS" => GetRestrictions(),
+            "RESERVEDWORDS" => GetReservedWords(connection),
+            "TABLES" => GetTables(connection, restrictionValues),
+            "COLUMNS" => GetColumns(connection, restrictionValues),
+            "FOREIGNKEYS" => GetForeignKeys(connection, restrictionValues),
+            "INDEXES" => GetIndexes(connection, restrictionValues),
+            _ => throw new ArgumentOutOfRangeException(nameof(collectionName), collectionName, "Invalid collection name.")
+        };
+
+    private static DataTable GetMetaDataCollections() =>
+        new(DbMetaDataCollectionNames.MetaDataCollections)
+        {
+            Columns =
+            {
+                { DbMetaDataColumnNames.CollectionName, typeof(string) },
+                { DbMetaDataColumnNames.NumberOfRestrictions, typeof(int) },
+                { DbMetaDataColumnNames.NumberOfIdentifierParts, typeof(int) }
+            },
+            Rows =
+            {
+                { DbMetaDataCollectionNames.MetaDataCollections, 0, 0 },
+                { DbMetaDataCollectionNames.Restrictions, 0, 0 },
+                { DbMetaDataCollectionNames.ReservedWords, 0, 0 },
+                { DuckDbMetaDataCollectionNames.Tables, 4, 3 },
+                { DuckDbMetaDataCollectionNames.Columns, 4, 4 },
+                { DuckDbMetaDataCollectionNames.ForeignKeys, 4, 3 },
+                { DuckDbMetaDataCollectionNames.Indexes, 4, 3 },
+            }
+        };
+
+    private static DataTable GetRestrictions() =>
+        new(DbMetaDataCollectionNames.Restrictions)
+        {
+            Columns =
+            {
+                { "CollectionName", typeof(string) },
+                { "RestrictionName", typeof(string) },
+                { "RestrictionDefault", typeof(string) },
+                { "RestrictionNumber", typeof(int) }
+            },
+            Rows =
+            {
+                { DuckDbMetaDataCollectionNames.Tables, "Catalog", "table_catalog", 1 },
```

Comment:
Can you create constants for these too? Similar to `DuckDbMetaDataCollectionNames`